### ISSUE TITLE
[Development] Letters: Check Profile Address Against Corp DB

### ISSUE THREAD

**Issue by luke-gcio** [department-of-veterans-affairs/va.gov-team#7816]
## User Story or Problem Statement
Problem description: We keep seeing errors related to address mismatch with VA profile and Corp DB.  We should be able to provocatively check for this mismatch and possibly  instruct the veteran to call the Call Center to have their Corp BD profile address updated to match their VA Profile.  This is related to the work for #7701.

## Goal
Give the veteran an option to fix the issue and not simply fail to obtain the letters.

## Acceptance Criteria
- [x] Not getting to the very end before we just fail - provide the veteran an option
---
## How to configure this issue
- [x] **Attached to a Milestone** (when will this be completed?)
- [x] **Attached to an Epic** (what body of work is this a part of?)
- [x] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [x] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [x] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by annaswims** [comment]
Sentry error http://sentry.vfs.va.gov/vets-gov/platform-api-production/issues/154347/?query=addressLine1

**Comment by annaswims** [comment]
A description of what the user is doing to see the error is in https://github.com/department-of-veterans-affairs/va.gov-team/issues/8575#issue-609247664

**Comment by edmangimelli** [comment]
@annaswims 
this step synchronized VA profile and Corp DB?
> I went to https://www.va.gov/profile/ and saw that I already had a valid address, clicked edit and save (did not change any data) and returned to https://www.va.gov/records/download-va-letters/letters/confirm-address and now I see

from [your closed ticket that you linked above](https://github.com/department-of-veterans-affairs/va.gov-team/issues/8575#issue-609247664)

**Comment by annaswims** [comment]
I know enough to describe the problem, but I don’t know what the path forward is. I think the root cause is that the address isn’t consistently syncing between CORPDB and VA Profile (the one formerly known as vets360, not the VA profile page of va.gov).  That seems like a hard problem to solve. But we can ask them to re-save their address, or maybe just re-save it on their behalf.  But asking them to re-save their address even if it's unchanged is weird/awkward.

**Comment by annaswims** [comment]
Apparently we've just hit the third anniversary of this bug. https://github.com/department-of-veterans-affairs/vets-api/pull/1731/files

It looks like we've got years worth of logged people affected in the database in InvalidLetterAddressEdipi. > 50k people affected.

**Comment by annaswims** [comment]
this comment seems rather relevant.

https://github.com/department-of-veterans-affairs/vets.gov-team/issues/11220#issuecomment-407845710

**Comment by annaswims** [comment]
google analytics for https://analytics.google.com/analytics/web/#/report/content-event-pages/a50123418w177519031p176188361/explorer-table.plotKeys=%5B%5D&_r.drilldown=analytics.pagePath:www.va.gov~2Frecords~2Fdownload-va-letters~2Fletters~2Fconfirm-address~2F&explorer-segmentExplorer.segmentId=analytics.eventLabel/

**Comment by annaswims** [comment]
I think this could be a less-complex front-end issue... or at least in part.   

We need EVSS to have a valid mailing address in order to generate the letter. We ask the user to verify their address on https://www.va.gov/records/download-va-letters/letters/confirm-address and give them the opportunity to edit their mailing address BUT the api call to https://api.va.gov/v0/letters is made before the user edits the address and no additional call is made aferwards.  

Also of note: The address gets updated in VA Profile, which syncs with CORPDB and when we make the call to EVSS for the letter EVSS is looking at CORPDB for the letter address (I think). We should verify this and how long this takes. 

@lihanli do you know how long it takes for an address to sync from VA Profile to CORPDB or if EVSS still connects to CORPDB and not VA Profile for addresses?

**Comment by annaswims** [comment]
So... td;dr for the front end... don't call https://api.va.gov/v0/letters until the address is verified or updated and if the address is updated we might want to wait a bit to try https://api.va.gov/v0/letters

**Comment by annaswims** [comment]
I think I'd try the above first, but if we still see errors after that we can check the address that EVSS has at https://api.va.gov/v0/address before a veteran continues.  The mailing address from v0/address  _should_ be exactly the same as the mailing address that we get from the v0/user user endpoint (which comes from VA profile), assuming the syncing works (but maybe it doesn't always).

**Comment by annaswims** [comment]
related support issue #6261

**Comment by annaswims** [comment]
I was hoping this would fix the problem, but it didn't (or at least not entirely).  I'll do some digging.

**Comment by annaswims** [comment]
I take it back.  I see a major improvement in google analytics. Deployed 3 pm feb 19
https://analytics.google.com/analytics/web/#/report/content-event-pages/a50123418w177519031p176188361/_u.date00=20210103&_u.date01=20210222&explorer-table.plotKeys=%5B%5D&explorer-table.filter=letter-list-failure&_r.drilldown=analytics.pagePath:www.va.gov~2Frecords~2Fdownload-va-letters~2Fletters~2Fconfirm-address~2F,analytics.eventAction:Letter%20-%20Failure&explorer-graphOptions.selected=analytics.nthWeek/

Yesterday we saw 7 errors.  On a typical day we saw hundreds

<img width="1301" alt="Screen Shot 2021-02-23 at 4 09 37 PM" src="https://user-images.githubusercontent.com/19188/108908706-9f976780-75f1-11eb-88e7-ec97e6ec652b.png">

**Comment by Mottie** [comment]
<img alt="ghe-emoji" title=":_badger:" src="https://cloud.githubusercontent.com/assets/136959/13512240/15d31412-e15f-11e5-8577-06cff0976977.gif"><img alt="ghe-emoji" title=":_badger:" src="https://cloud.githubusercontent.com/assets/136959/13512240/15d31412-e15f-11e5-8577-06cff0976977.gif"><img alt="ghe-emoji" title=":_badger:" src="https://cloud.githubusercontent.com/assets/136959/13512240/15d31412-e15f-11e5-8577-06cff0976977.gif"><img alt="ghe-emoji" title=":_badger:" src="https://cloud.githubusercontent.com/assets/136959/13512240/15d31412-e15f-11e5-8577-06cff0976977.gif"> 

<img alt="ghe-emoji" title=":_fist_bump:" src="https://cloud.githubusercontent.com/assets/136959/13559365/0fe4e2a2-e3d9-11e5-9b10-32de5d5988a2.gif">

**Comment by annaswims** [comment]
I made https://github.com/department-of-veterans-affairs/va.gov-team/issues/21066 as a follow-up issue to this issue.  We can debate whether https://github.com/department-of-veterans-affairs/vets-api/pull/6099 should be merged before or after #21066 is complete at standup today. 


pro's of merging 6099 sooner: the api request was going to fail anyway and we'll have more data
cons 🤷‍♀️. The error message is inaccurate, but still just as inaccurate.
